docs/development/setup-advanced.md

```diff
@@ -55,6 +56,76 @@ on using the Zulip development
 environment](../development/setup-vagrant.html#step-4-developing),
 ignoring the parts about `vagrant` (since you're not using it).
 
+## Installing directly on Windows 10 (Experimental)
+
+We will be using Microsoft's new feature [WSL 2](https://docs.microsoft.com/en-us/windows/wsl/wsl2-about)
+for installation. So, first install WSL 2 by following the instructions provided by Microsoft
+[here](https://docs.microsoft.com/en-us/windows/wsl/wsl2-install). Install `Ubuntu 18.04` as your linux
+distribution from the Microsoft Store.
+
+Launch `Ubuntu 18.04` shell and run the following commands
+(click `Allow access` if you get popups for windows firewall blocking some services):
+
+```
+sudo apt update && sudo apt upgrade
+sudo apt install rabbitmq-server memcached redis-server postgresql rabbitmq-server
+```
+
+Open `/etc/rabbitmq/Rabbitmq.conf` file using:
+
+```
+sudo vim /etc/rabbitmq/Rabbitmq.conf
+```
+
+Add the following lines at the end of your file:
+
+```
+NODE_IP_ADDRESS=127.0.0.1
+NODE_PORT=5672
+```
+
+Start the servers:
+
+```
+sudo service rabbitmq-server start && sudo service memcached start &&\
+sudo service redis-server start && sudo service postgresql start
+```
+
+Then move to [cloning your fork of the Zulip repository][zulip-rtd-git-cloning]
+and [connecting the Zulip upstream repository][zulip-rtd-git-connect]:
+
+```
+git clone --config pull.rebase https://github.com/YOURUSERNAME/zulip.git
+cd zulip
+git remote add -f upstream https://github.com/zulip/zulip.git
+```
+
+Now run:
+
+```
+./tools/provision
+source /srv/zulip-py3-venv/bin/activate
+./tools/run-dev.py  # starts the development server
+```
+
+Please note that if you shutdown WSL, you will have to manually start
+some database services.
+This can be easily done by adding
+
+```
+alias start_servers=sudo service \
+rabbitmq-server start && sudo service memcached start && sudo service \
+redis-server start && sudo service postgresql start
```

Comment:
Why don't we make this a `tools/wsl/start-services` script?  That feels more maintainable.

Reply:
(I'm not sure we want to ask folks to put it in a `bashrc`; that seems messy.

Reply:
(And we can just mention that same script above).

Reply:
Thanks for the review! Updated with new changes.